src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -77,7 +77,49 @@ public Object deletemenuitem(
         
          driverAvailabilityRepository.delete(drivav);
         return genericMessage("Driver Availability with id %s deleted".formatted(id));
-        }
+        }    
+
+    @Operation(summary= "Update Driver Availability")
+    @PreAuthorize("hasAnyRole('ADMIN', 'DRIVER')")
+    @PutMapping("")
+    public DriverAvailability updatemenuitem(
+            @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid DriverAvailability incoming) {
+
+        DriverAvailability drivav = driverAvailabilityRepository.findById(id)
+        .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+
+        drivav.setDriverId(incoming.getDriverId());
+        drivav.setDay(incoming.getDay());
+        drivav.setStartTime(incoming.getStartTime());
+        drivav.setEndTime(incoming.getEndTime());
+        drivav.setNotes(incoming.getNotes());
+
+        driverAvailabilityRepository.save(drivav);
+
+        return drivav;
+    }
+
+
+    @Operation(summary= "List all Driver Availability")
+    @PreAuthorize("hasAnyRole('ADMIN', 'DRIVER','USER')")
+    @GetMapping("/all")
+    public Iterable<DriverAvailability> allmenuitems() {
+        Iterable<DriverAvailability> drivall = driverAvailabilityRepository.findAll();
+        return drivall;
+    }
+
+
+    @Operation(summary= "Get Driver Availability")
+    @PreAuthorize("hasAnyRole('ADMIN', 'DRIVER','USER')")
+    @GetMapping("")
```

Comment:
Regular users should not be able to view driver availabilities, could be pontentially harmful

Other than that good to merge, ping me after change